src/Microsoft.DocAsCode.Metadata.ManagedReference/ExtractMetadataWorker.cs

```diff
@@ -402,26 +469,27 @@ private static void CopyFromCachedResult(BuildInfo buildInfo, IEnumerable<string
             Logger.Log(LogLevel.Info, $"'{inputs.ToDelimitedString()}' keep up-to-date since '{buildInfo.TriggeredUtcTime.ToString()}', cached result from '{buildInfo.OutputFolder}' is used.");
             relativeFiles.Select(s => Path.Combine(outputFolderSource, s)).CopyFilesToFolder(outputFolderSource, outputFolder, true, s => Logger.Log(LogLevel.Info, s), null);
         }
-        
-        private static Task<MetadataItem> GetProjectMetadataFromCacheAsync(Project project, string outputFolder, ProjectDocumentCache documentCache, bool forceRebuild, bool preserveRawInlineComments, string filterConfigFile)
+
+        private static Task<MetadataItem> GetProjectMetadataFromCacheAsync(Project project, Compilation compilation, string outputFolder, ProjectDocumentCache documentCache, bool forceRebuild, bool preserveRawInlineComments, string filterConfigFile, IEnumerable<IMethodSymbol> extensionMethods)
         {
             var projectFilePath = project.FilePath;
             var k = documentCache.GetDocuments(projectFilePath);
             return GetMetadataFromProjectLevelCacheAsync(
                 project,
                 new[] { projectFilePath, filterConfigFile },
                 s => Task.FromResult(forceRebuild || s.AreFilesModified(k.Concat(new string[] { filterConfigFile }))),
-                s => project.GetCompilationAsync(),
+                s => Task.FromResult(compilation),
```

Comment:
Project cache might not work.
If we have project X with class A, and no changes, but we have project Y with class AExtension, and it updated.
Then we can't use the cache for project X, but we can't detected the cache for project X is expired.